Mathport/Syntax/Translate/Tactic/Mathlib/RCases.lean

```diff
@@ -49,20 +49,19 @@ end
 
 @[trTactic obtain] def trObtain : TacM Syntax := do
   let ((pat, ty), vals) ← parse obtainArg
-  liftM $ show M _ from
-    return mkNode ``Parser.Tactic.obtain #[mkAtom "obtain",
-      mkOptionalNode (← pat.mapM trRCasesPatMed),
-      ← mkOptionalNodeM ty fun ty => return #[mkAtom ":", ← trExpr ty],
-      ← mkOptionalNodeM vals fun vals =>
-        return #[mkAtom ":=", (mkAtom ",").mkSep $ ← vals.mapM trExpr]]
+  -- liftM $ show M _ from
+  let vals ← vals.mapM fun vals => vals.mapM (liftM do trExpr ·)
+  liftM do `(tactic|
+    obtain $[$(← pat.mapM trRCasesPatMed):rcasesPatMed]?
+      $[: $(← ty.mapM trExpr)]? $[:= $[$vals],*]?)
```

Comment:
trExpr should work left to right in lean 3 source order
```suggestion
  let pat ← pat.mapM trRCasesPatMed
  let ty ← ty.mapM trExpr
  let vals ← vals.mapM fun vals => vals.mapM (liftM do trExpr ·)
  liftM do `(tactic| obtain $[$pat:rcasesPatMed]? $[: $ty]? $[:= $[$vals],*]?)
```

Reply:
Wait, why?  I'm pretty sure I've never paid attention to this, so I wouldn't be surprised if we accidentally violated this somewhere.

Reply:
I don't think it makes a big difference, but I think there are some things that assume that elaboration goes in source order like the comment inserter.

Reply:
The comment inserter doesn't care at all about the order `trExpr` is called in.  It uses the `SourceInfo` from the translated `Syntax`.  And `trExpr` sets this `SourceInfo` based in the `Meta` data in the `Spanned Expr` argument.